api/v1/slo_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SloSpec defines the desired state of Slo
+type SloSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of Slo. Edit slo_types.go to remove/update
+	Name        string            `json:"name"`
+	Description string            `json:"description,omitempty"`
+	Type        string            `json:"type"`
+	Window      string            `json:"window"`
+	Slo         int               `json:"slo"`
+	Threshold   int               `json:"threshold"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+}
+
+// SloStatus defines the observed state of Slo
+type SloStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	SloStatus string `json:"SloStatus"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
```

Comment:
i would also like to add a shortname using kubebuilder. think about a good one :)

Reply:
can't think about a shorter-than-3-letters name. We can use `objective`, but it's longer than `slo`. `Obj` for "objective" is equal but most people would probably think it's "object", which would be confusing.